chia/util/keychain.py

```diff
@@ -87,11 +87,32 @@ def bytes_to_mnemonic(mnemonic_bytes: bytes) -> str:
     return " ".join(mnemonics)
 
 
-def bytes_from_mnemonic(mnemonic_str: str) -> bytes:
+def mnemonic_from_short_words(mnemonic_str: str) -> str:
+    """
+    Since the first 4 letters of each word is unique (or the full word, if less than 4 characters), and its common
+    practice to only store the first 4 letters of each word in many offline storage solutions, also support looking
+    up words by the first 4 characters
+    """
     mnemonic: List[str] = mnemonic_str.split(" ")
     if len(mnemonic) not in [12, 15, 18, 21, 24]:
         raise ValueError("Invalid mnemonic length")
 
+    four_char_dict = {word[: min(4, len(word))]: word for word in bip39_word_list().splitlines()}
+    full_words: List[str] = []
+    for i in range(0, len(mnemonic)):
+        word = mnemonic[i]
+        full_word = four_char_dict.get(word[: min(4, len(word))])
+        if not full_word:
+            raise ValueError(f"'{word}' is not in the mnemonic dictionary; may be misspelled")
```

Comment:
Let python format it in a quoted form.  For example, if there's a `'` in there then the existing quoting would be odd.  The `r` is for `repr()` rather than `str()` which is used otherwise.

```suggestion
            raise ValueError(f"{word!r} is not in the mnemonic dictionary; may be misspelled")
```